### PR TITLE
Add overload to handle the case when the user writes:

### PR DESCRIPTION
**Pull request by schweitzpgi**
```c++
  cudaq::control(Kernel{}, {ctrl1, ctrl2}, args...);
```
Get vector<reference_wrapper<qubit>> through the bridge.

Add simple test.

### CODE REVIEW

File: test/AST-Quake/control.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+// RUN: cudaq-quake %s | FileCheck %s
+
+#include <cudaq.h>
+
+struct heisenbergU {
+  void operator()(cudaq::qreg<> &q) __qpu__ {
+    auto nQubits = q.size();
+    for (int step = 0; step < 100; ++step) {
+      // Fixme need a way to apply ar Rn to all qubits
```
Comment (amccaskey) [inline test/AST-Quake/control.cpp:17]: You could probably remove this comment - just me talking to myself :)

**Comment by amccaskey** [comment]
Will you check that the following makes it through the bridge with this PR:

```cpp

struct givens {
  void operator()(double lambda, cudaq::qubit &q, cudaq::qubit &r) __qpu__ {
    ry(M_PI_2, q);
    ry(M_PI_2, r);
    z<cudaq::ctrl>(q, r);
    ry(lambda, q);
    ry(-lambda, r);
    z<cudaq::ctrl>(q, r);
    ry(-M_PI_2, q);
    ry(-M_PI_2, r);
  }
};

__qpu__ void qnppx(double theta, cudaq::qubit &q, cudaq::qubit &r,
                   cudaq::qubit &s, cudaq::qubit &t) {
  x<cudaq::ctrl>(r, q);
  x<cudaq::ctrl>(s, t);
  cudaq::control(givens{}, {q, t}, theta, r, s);
  x<cudaq::ctrl>(r, q);
  x<cudaq::ctrl>(s, t);
}
```

**Comment by schweitzpgi** [comment]
> Will you check that the following makes it through the bridge with this PR:
> 
> ```c++
> struct givens {
>   void operator()(double lambda, cudaq::qubit &q, cudaq::qubit &r) __qpu__ {
>     ry(M_PI_2, q);
>     ry(M_PI_2, r);
>     z<cudaq::ctrl>(q, r);
>     ry(lambda, q);
>     ry(-lambda, r);
>     z<cudaq::ctrl>(q, r);
>     ry(-M_PI_2, q);
>     ry(-M_PI_2, r);
>   }
> };
> 
> __qpu__ void qnppx(double theta, cudaq::qubit &q, cudaq::qubit &r,
>                    cudaq::qubit &s, cudaq::qubit &t) {
>   x<cudaq::ctrl>(r, q);
>   x<cudaq::ctrl>(s, t);
>   cudaq::control(givens{}, {q, t}, theta, r, s);
>   x<cudaq::ctrl>(r, q);
>   x<cudaq::ctrl>(s, t);
> }
> ```

Thanks for the test. I'm getting a failure on this code. Investigating.

**Comment by amccaskey** [comment]
Thanks for the bug fix, the code snippet works for me now.

**Review by amccaskey — APPROVED** [review]
Approved pending CI

**Comment by bettinaheim** [comment]
Rel notes: see also https://github.com/NVIDIA/cuda-quantum/pull/35.

**Pull request merged**